data/eos-save-icon-grid.in

```diff
@@ -86,15 +90,109 @@ const getLocale = function(args) {
     return locale;
 }
 
-let locale = getLocale(ARGV);
+const writeDirectoryDesktop = function(directoryId, name) {
+    // Save a .desktop file for the given icon grid directory. This contains
+    // metadata for the directory. It should be copied into
+    // `/usr/local/share/desktop-directories` on the image, typically using a
+    // hook like
+    // https://github.com/endlessm/endless-image-config/blob/c03674a7/hooks/image/51-hack-desktop-folders.chroot
+
+    const path = GLib.build_filenamev([SAVE_PATH, directoryId + '.desktop']);
+
+    const contents = `[Desktop Entry]
+Version=1.0
+Name${(locale != 'C') ? '[' + locale + ']' : ''}=${name}
+Type=Directory`;
+
+    GLib.file_set_contents(path, contents);
+
+    return path
+}
+
+const generateJsonAndDesktops = function(locale) {
+    // Save a JSON file for the icon grid layout. This describes the icon grid
+    // layout in a way which will be loaded by eos-desktop-extension (see
+    // https://github.com/endlessm/eos-desktop-extension/blob/f6fa4d24/settings.js#L124)
+    // and eventually stored in the new-style GSettings keys for the icon grid
+    // configuration. See:
+    // ```
+    // gsettings get org.gnome.shell app-picker-layout
+    // gsettings get org.gnome.desktop.app-folders folder-children
+    // gsettings list-recursively org.gnome.desktop.app-folders.folder:/org/gnome/desktop/app-folders/folders/${folderId}/
+    // ```
+    //
+    // The JSON file has type `a{sas}`, where each dictionary key is a folder ID
+    // which maps to an array of desktop file names for the icons in that
+    // directory. The `desktop` key is special, and means the main icon grid.
+    //
+    // Each folder ID corresponds to a `.desktop` file in
+    // `/usr/local/share/desktop-directories` which gives the metadata (mostly,
+    // the human readable name) for that folder. See `writeDirectoryDesktop()`.
+    let jsonOut = {
```

Comment:
I believe this one can be `const` too - even though the dict itself is modified, you don't assign it to anything else, so it still qualifies as const. Same for the `filesWritten` variable below.